### Mesclando datasets

In [1]:
import pandas as pd

In [3]:
parte1 = pd.read_json('parte1.json')

In [4]:
parte1.to_csv("parte1.csv",index=False)

In [6]:
parte2 = pd.read_csv('parte2.csv', sep=',')

In [7]:
import sqlite3
import pandas as pd
# Create your connection.
cnx = sqlite3.connect('parte3.sqlite')

parte3 = pd.read_sql_query("SELECT * FROM parte3 WHERE tipo_dado = 1", cnx)

In [8]:
parte3 = parte3[['year','sex','name','n','prop']]

In [ ]:
parte3.to_csv("parte2.csv",index=False)

In [9]:
dataframe = pd.concat([parte1, parte2, parte3], axis=0)

In [ ]:
dataframe.to_csv("data.csv",index=False)

In [10]:
dataframe = pd.read_csv('data.csv', sep='^')

year sex      name     n      prop
0       1880   F      Mary  7065  0.072400
1       1880   F     Sarah  1288  0.013200
2       1880   F     Clara  1226  0.012600
3       1880   F      Ella  1156  0.011800
4       1880   F  Florence  1063  0.010900
...      ...  ..       ...   ...       ...
641550  2017   M     Zevon     5  0.000003
641551  2017   M    Zikora     5  0.000003
641552  2017   M    Zubeyr     5  0.000003
641553  2017   M    Zyheem     5  0.000003
641554  2017   M     Zykai     5  0.000003

[1924665 rows x 5 columns]

### 1- Quantos nomes diferentes existem por ano a partir de 2000? Forneça uma tabela e demonstre os resultados graficamente.

In [11]:
df_posterior_2000 = dataframe.loc[(dataframe.year > 2000)]

In [14]:
df_posterior_2000

year sex     name      n      prop
454098  2001   F  Madison  22164  0.011200
454099  2001   F   Ashley  16526  0.008300
454100  2001   F   Olivia  13978  0.007100
454101  2001   F     Emma  13307  0.006700
454102  2001   F   Alyssa  13269  0.006700
...      ...  ..      ...    ...       ...
641550  2017   M    Zevon      5  0.000003
641551  2017   M   Zikora      5  0.000003
641552  2017   M   Zubeyr      5  0.000003
641553  2017   M   Zyheem      5  0.000003
641554  2017   M    Zykai      5  0.000003

[562156 rows x 5 columns]

In [12]:
year = df_posterior_2000.groupby('year')['name']

In [15]:
pd.DataFrame(year.describe().unique)

unique
year       
2001  27980
2002  28279
2003  28886
2004  29501
2005  30153
2006  31624
2007  32416
2008  32510
2009  32242
2010  31623
2011  31449
2012  31279
2013  30834
2014  30731
2015  30583
2016  30386
2017  29910

### 2 - Qual a média e a mediana da contagem de bebês no dataset. Qual dessas medidas de tendência central você escolheria para descrever esse dado, justifique sua opção.

In [16]:
df_media = dataframe.n.mean()
df_media

180.87330366583276

In [17]:
df_mediana = dataframe.n.median()
df_mediana

12.0

Optaria por usar a mediana, pois os valores da contagem de bebês (n) é muito discrepante.

### 3- Qual a média e desvio padrão da contagem de bebês no ano de 1997?


In [19]:
df_1997 = dataframe.loc[(dataframe.year == 1997)]
df_1997.describe()

year             n          prop
count  26971.0  26971.000000  26971.000000
mean    1997.0    134.396166      0.000068
std        0.0   1024.324445      0.000521
min     1997.0      5.000000      0.000000
25%     1997.0      6.000000      0.000003
50%     1997.0     11.000000      0.000005
75%     1997.0     28.000000      0.000010
max     1997.0  37548.000000      0.018799

A média é 134.39 e o desvio padrão é 1024.32

### 4- Levando em conta que o dataset engloba o nascimento de todos os bebês do país imaginário Hilablândia. Qual o total de nascimentos no ano de 2002? Desses, quantos são do sexo feminino e quantos do sexo masculino?

In [20]:
df_2002 = dataframe.loc[(dataframe.year == 2002)]
df_2002.describe()

year             n          prop
count  30563.0  30563.000000  30563.000000
mean    2002.0    122.240683      0.000059
std        0.0    877.080094      0.000432
min     2002.0      5.000000      0.000000
25%     2002.0      7.000000      0.000003
50%     2002.0     11.000000      0.000005
75%     2002.0     28.000000      0.000010
max     2002.0  30568.000000      0.014800

In [21]:
df = df_2002.groupby('sex')['n'].describe()
df

count        mean          std  min  25%   50%   75%      max
sex                                                                 
F    18081.0   99.316465   658.737325  5.0  7.0  11.0  27.0  24463.0
M    12482.0  155.447925  1119.479065  5.0  7.0  11.0  29.0  30568.0

Foram 30563 bebes nascidos no ano de 2002. Foram 18081 bebes femininos e 12482 bebes masculinos

### 5 - Descobri qual é nome mais utilizado por ano desde do ano 2000

In [22]:
data_1 = dataframe.loc[(dataframe.year > 2000)]

In [23]:
year_1 = data_1.groupby('year')['name']

In [24]:
pd.DataFrame(year_1.describe().top)

top
year          
2001   Madison
2002    Ashley
2003      Emma
2004    Olivia
2005      Emma
2006     Emily
2007       Ava
2008      Emma
2009  Isabella
2010  Isabella
2011  Isabella
2012  Isabella
2013    Sophia
2014  Isabella
2015      Emma
2016  Isabella
2017    Amelia

In [25]:
top = pd.DataFrame(year_1.describe().top)

In [26]:
top.describe()

top
count         17
unique         9
top     Isabella
freq           6